pkg/spoke/addon/registration_controller.go

```diff
@@ -74,11 +79,44 @@ func NewAddOnRegistrationController(
 			},
 			hubAddOnInformers.Informer()).
 		WithSync(c.sync).
+		ResyncEvery(10*time.Minute).
 		ToController("AddOnRegistrationController", recorder)
 }
 
 func (c *addOnRegistrationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	addOnName := syncCtx.QueueKey()
+	queueKey := syncCtx.QueueKey()
+	if queueKey != factory.DefaultQueueKey {
+		// sync a particular addOn
+		return c.syncAddOn(ctx, syncCtx, queueKey)
+	}
+
+	// handle resync
+	addOns, err := c.hubAddOnLister.ManagedClusterAddOns(c.clusterName).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	errs := []error{}
+	for _, addOn := range addOns {
+		_, err := c.hubAddOnClient.AddonV1alpha1().ManagedClusterAddOns(c.clusterName).Get(ctx, addOn.Name, metav1.GetOptions{})
```

Comment:
why not get from cache?

Reply:
Code updated to sync between cached registration configs and ManagedClusterAddOns in informer cache.